### PR TITLE
Esp idf v5 fixes

### PR DESCRIPTION
**Pull request by bandi13**
# Description

Using the latest ESP-IDF environment there were some changes to the library and thus our stuff wouldn't compile out of the box.

### CODE REVIEW

**Review by gojimmypi — CHANGES_REQUESTED** [review]
I might have formatted this:

```
    double current_time(int reset)
    {
        #if ESP_IDF_VERSION_MAJOR >= 4
        TickType_t tickCount;
        #else
        portTickType tickCount;
        #endif
```
like this instead for readability:

```
    double current_time(int reset)
    {
#if ESP_IDF_VERSION_MAJOR >= 4
        TickType_t tickCount;
#else
        portTickType tickCount;
#endif
```
There should probably be something in the coding standards document for `#define` position and indenting. Generally they seem to always be fully left-justified unless (sometimes) when nested.

Also - this does seem to work for me in ESP-IDF v4.4.2; You've tested with v5?

**Comment by gojimmypi** [comment]
Heads up your [bandi13:ESP-IDF_fixes](https://github.com/bandi13/wolfssl/tree/ESP-IDF_fixes) branch is a bit behind master and there are a couple of potentially interesting / related merge PR's pending. See https://github.com/wolfSSL/wolfssl/pull/5871 and recently-merged https://github.com/wolfSSL/wolfssl/pull/5903  and https://github.com/wolfSSL/wolfssl/pull/5646

I might have given better commit names than "better fixes" (e.g. actually call out the version-check `TaskHandle_t` vs `xTaskHandle` changes).

Otherwise nice update that seems to work for me in v4.4! Did you encounter these problems in ESP-IDF5?

See also this curiosity: https://github.com/wolfSSL/wolfssl/issues/5727 as well as https://github.com/wolfSSL/wolfssl/issues/5319 (the related `autogen` question in https://github.com/wolfSSL/wolfssl/issues/5883 ) and https://github.com/wolfSSL/wolfssl/issues/5215

**Comment by gojimmypi** [comment]
I tested this on ESP-IDF v4.4.2 and confirmed the benchmark and tests run to completion without error:

```
git clone https://github.com/bandi13/wolfssl.git bandi13-wolfssl-install --branch ESP-IDF_fixes --depth 1
. /mnt/c/SysGCC/esp32/esp-idf/v4.4.2/export.sh
cd bandi13-wolfssl-install/IDE/Espressif/ESP-IDF
./setup.sh
cd examples/wolfssl_benchmark
idf.py -p /dev/ttyS20 -b 115200 build flash monitor

# exit with Ctrl-[
cd ../wolfssl_test
idf.py -p /dev/ttyS20 -b 115200 build flash monitor
```

Note there's a warning in ESP-IDF v5.0 in https://github.com/wolfSSL/wolfssl/issues/5909 about duplicate definitions in `user_settings.h`. It may be specific to the docker build. More work is needed to verify Espressif ESP-IDF v5 is fully supported in all of the ESP32 wolfSSL examples.

Although my change request has been implemented, my approval here is not authoritative and someone such as @dgarske should review and actually perform the merge.

**Review by gojimmypi — APPROVED** [review]

**Review by dgarske — APPROVED** [review]

**Pull request merged**
